### DISCUSSION TITLE
Date Time Format

### DISCUSSION THREAD

**Discussion by DrNatsu** [apache/superset#30053] [Q&A / Help]
I'm encountering an issue with Apache Superset when working with a TIME data type that includes microseconds. In my database, I have a TIME column (departure_time) that is correctly stored and displayed in the format HH:MM:SS.ffffff. However, when I attempt to use this column in Superset, I'm seeing unexpected behaviors such as NaN values and incorrect timestamp interpretations.

Specifically:

1. Superset recognizes the column as TIME, but attempts to specify a format (e.g., %H:%M:%S) in the "Datetime Format" field result in an "Invalid date/timestamp format" error.
2. Even without specifying a format, the column does not render correctly in visualizations.
3. Casting the TIME to a simpler format or VARCHAR resolves the issue, but this seems like a workaround rather than a solution.

**Comment by gvvsss** [comment]
Hi, 
I am also facing similar issue, on Superset 4.1.2, Postgres 17. I have a 'time' column in a table, which is showing fine in SQL Lab and in Sample/Result section of the chart. But, in the actual chart the value is being displayed as NaN:Nan:Nan. 

It is only a time column (time in HH:mm:ss format, no milliseconds) and is identified as such correctly in Superset, only the display on chart is faulty.
